lib/blocks/categories.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Server-side rendering of the `core/categories` block.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Renders the `core/categories` block on server.
+ *
+ * @param array $attributes The block attributes.
+ *
+ * @return string Returns the categories list/dropdown markup.
+ */
+function gutenberg_render_block_core_categories( $attributes ) {
+	$align = 'center';
+	if ( isset( $attributes['align'] ) && in_array( $attributes['align'], array( 'left', 'right', 'full' ), true ) ) {
+		$align = $attributes['align'];
+	}
+
+	$args = array(
+		'echo'         => false,
+		'hierarchical' => ! empty( $attributes['showHierarchy'] ),
+		'orderby'      => 'name',
+		'show_count'   => ! empty( $attributes['showPostCounts'] ),
+		'title_li'     => '',
+	);
+
+	if ( ! empty( $attributes['displayAsDropdown'] ) ) {
+		$id = 'wp-block-categories-' . wp_rand();
```

Comment:
Makes sense. I've updated it to use an increasing static counter, see c501a4bbc4b1fbfc52d8505a20619a4367d73eb6